sdks/java/io/google-cloud-platform/src/main/java/org/apache/beam/sdk/io/gcp/bigquery/BigQueryAvroUtils.java

```diff
@@ -311,12 +304,10 @@ private static Object convertRequiredField(
         verify(v instanceof Boolean, "Expected Boolean, got %s", v.getClass());
         return v;
       case "TIMESTAMP":
-        // TIMESTAMP data types are represented as Avro LONG types. They are converted back to
-        // Strings with variable precision (up to six digits) to match the JSON files exported by
-        // BigQuery.
+        // TIMESTAMP data types are represented as Avro LONG types, microseconds since the epoch.
+        // Values may be negative since BigQuery timestamps start at 0001-01-01 00:00:00 UTC.
         verify(v instanceof Long, "Expected Long, got %s", v.getClass());
-        double doubleValue = ((Long) v) / 1_000_000.0;
-        return formatTimestamp(Double.toString(doubleValue));
+        return formatTimestamp((Long) v);
```

Comment:
Maybe you don't need to cast it if you've checked its type?

Reply:
I had to cast that explicitly; `v` is of type `Object`.